generate/genhcl/genhcl.go

```diff
@@ -239,6 +254,16 @@ func validateGenerateHCLBlock(block *hclsyntax.Block) error {
 	if len(block.Body.Attributes) != 0 {
 		return errors.New("attributes are not allowed")
 	}
+	if len(block.Body.Blocks) != 1 {
+		return fmt.Errorf("one 'content' block is required, got %d blocks", len(block.Body.Blocks))
+	}
+	contentBlock := block.Body.Blocks[0]
+	if contentBlock.Type != "content" {
+		return fmt.Errorf("one 'content' block is required, got %q block", contentBlock.Type)
+	}
+	if len(contentBlock.Labels) > 0 {
+		return fmt.Errorf("content block has unexpected labels: %v", contentBlock.Labels)
+	}
```

Comment:
why do we manually check those... can't we use the go structures to let HCL library check this for us?

Reply:
When you mentioned "Go Structures" I immediately thought of [hclsimple](https://pkg.go.dev/github.com/hashicorp/hcl/v2/hclsimple).

Which in this case seems less helpful IMO (but quite helpful on some scenarios), but now I think maybe you meant the [BodySchema](https://pkg.go.dev/github.com/hashicorp/hcl2@v0.0.0-20191002203319-fb75b3253c80/hcl#BodySchema).

Which maybe could make some of the basic validation for us, will take a look (haven't used it yet, sorry for the ignorance).